keylime/migrations/versions/8a44a4364f5a_initial_database_migration.py

```diff
@@ -5,6 +5,9 @@
 Create Date: 2020-10-08 20:29:54.883816
 
 """
+
+#pylint: disable=no-member
+
```

Comment:
This is due to requirements not satisfied, I suggest we install necessary requirements into a virtualenv which the keylime can run, then these checking would make sense. Also using determined/ranged version in the requirements will help us find potential incompatible API issues from dependencies.

Reply:
I have the requirements on my system that is running and it is failing this command here and shows the errors as reported in the commit message:

```
python3 -m pylint --disable=all --enable=no-member *.py $(find ./keylime ./test -name '*.py' ! -name 'oldtest.py')
```

I also tried it with a virtualenv.

pylint shows this error here for example:

```
************* Module keylime.tpm.tpm1
keylime/tpm/tpm1.py:537:20: E1101: Module 'M2Crypto.m2' has no 'bn_to_mpi' member (no-member)
keylime/tpm/tpm1.py:537:33: E1101: Module 'M2Crypto.m2' has no 'hex_to_bn' member (no-member)
keylime/tpm/tpm1.py:538:20: E1101: Module 'M2Crypto.m2' has no 'bn_to_mpi' member (no-member)
keylime/tpm/tpm1.py:538:33: E1101: Module 'M2Crypto.m2' has no 'hex_to_bn' member (no-member)
```

But then the following shows that 'hex_to_bn' is available  in the virtualenv:

```
(env) stefanb@linux0e:~/keylime$ python3 -c "import M2Crypto.m2; print('hex_to_bn' in dir(M2Crypto.m2))"
True
```

It is also available in the host:

```
stefanb@linux0e:~/keylime$ python3 -c "import M2Crypto.m2; print('hex_to_bn' in dir(M2Crypto.m2))"
True
```

So what am I doing wrong?

Reply:
Ah, let me guess, the hex_to_bn is implemented in C, that's why pylint is unable to get those signatures. Hmm ..
I found this http://pylint.pycqa.org/en/latest/technical_reference/c_extensions.html that might be useful.

Reply:
> Ah, let me guess, the hex_to_bn is implemented in C, that's why pylint is unable to get those signatures. Hmm ..
> I found this http://pylint.pycqa.org/en/latest/technical_reference/c_extensions.html that might be useful.

Haven't had much luck with (all kinds of combinations of) these two command line options...

```
python3 -m pylint --disable=all --enable=no-member --unsafe-load-any-extension=y --extension-pkg-whitelist=zmq,M2Crypto.m2 *.py $(find ./keylime ./test -name '*.py' ! -name 'oldtest.py')
```

zmq and M2Crpto.m2 related errors still show up...

Reply:
```--ignored-modules``` solves the issue...